### ISSUE TITLE
Unable to use 'type' key for aggregation after transition to OpenSearch.Client

### ISSUE THREAD

**Issue by GenPike** [opensearch-project/opensearch-net#362]
### What is the bug?
Trying to add new key 'type' to `AggregationDictionary`, which throws `System.ArgumentException: 'type' is one of the reserved aggregation keywords we use a heuristics...`
That happened after our OpenSearch version was updated to 2.3, which obliged us switch to OpenSearch.Client 1.5 from NEST 7.10.0.

Seems like that's not really a bug, but just client behavior.
So key is validated in [ValidateKey](https://github.com/opensearch-project/opensearch-net/blob/main/src/OpenSearch.Client/Aggregations/AggregationContainer.cs#L85) method, whether keyword used.
However, I don't see this method override check in [NEST](https://github.com/elastic/elasticsearch-net/blob/main/src/Elastic.Clients.Elasticsearch/Core/Aggregations/AggregationDictionary.cs#L52).
This was added to OpenSearch.Client with [Commit](https://github.com/opensearch-project/opensearch-net/commit/588b1c672be4d219d39fe0b1c6fb62888cebd892#diff-7b108d1fc40044113396ebeadab0d3bb3d2b5f7b499ef543c758e634394facf6)

So my real question - is there any workaround how we can still use keyword without exception? 

### How can one reproduce the bug?
Doing simple Terms based aggregate request:
```
var aggregationDictionary = new AggregationDictionary();
aggregationDictionary.Add(
    "type",
    new TermsAggregation("")
    {
        Field = "data.type"
    });
```

### What is the expected behavior?
Well, no such exception, as before using NEST.

### What is your host/environment?
- OpenSearch.Client 1.5.0
- OpenSearch 2.3 running in AWS OpenSearch Service
- NEST 7.10.0

**Comment by Xtansia** [comment]
The specific exclusion of the `"type"` looks to have been included in [v7.11 of NEST](https://github.com/elastic/elasticsearch-net/pull/5286) and occurred pre-fork of the client. While it may be frustrating, the way to work around this is to rename your aggregate, ie. 
```csharp
var aggregationDictionary = new AggregationDictionary();
aggregationDictionary.Add(
    "data_type",
    new TermsAggregation("")
    {
        Field = "data.type"
    });
```

By the looks of it, as the error implies, the aggregation json parser would need complete re-writing to resolve this in general. It may have been resolved in v8+ of NEST but however due to licensing is not-applicable to OpenSearch.

The very specific case of `"type"` could maybe be resolved by reverting the support for the `geo_line` aggregate which as far as I'm aware doesn't exist in OpenSearch.

**Comment by GenPike** [comment]
Thank you for explained options! Rename approach is simple and acceptable, no need for any further refactoring steps..
